mpas_analysis/test/test_mpas_xarray.py

```diff
@@ -0,0 +1,130 @@
+                                                             yearoffset=1850))
+        ds = mpas_xarray.subset_variables(ds, varList)
+        self.assertEqual(ds.data_vars.keys(), varList)
+        self.assertEqual(pd.Timestamp(ds.Time.values[0]),
+                         pd.Timestamp('1855-01-16 12:22:30'))
+
+        # next, test the same with the onlyvars argument
+        ds = xr.open_mfdataset(
+            fileName,
+            preprocess=lambda x: mpas_xarray.preprocess_mpas(x,
+                                                             timestr=timestr,
+                                                             onlyvars=varList,
+                                                             yearoffset=1850))
+        self.assertEqual(ds.data_vars.keys(), varList)
+
+    def test_iselvals(self):
+        fileName = str(self.datadir.join('example_jan.nc'))
+        timestr = 'time_avg_daysSinceStartOfSim'
+        varList = \
+            ['time_avg_avgValueWithinOceanLayerRegion_avgLayerTemperature',
+             'refBottomDepth']
+
+        iselvals = {'nVertLevels': slice(0, 3)}
+        ds = xr.open_mfdataset(
+            fileName,
+            preprocess=lambda x: mpas_xarray.preprocess_mpas(x,
+                                                             timestr=timestr,
+                                                             onlyvars=varList,
+                                                             iselvals=iselvals,
+                                                             yearoffset=1850))
+        self.assertEqual(ds.data_vars.keys(), varList)
+        self.assertEqual(ds[varList[0]].shape, (1, 7, 3))
+        self.assertEqual(ds['refBottomDepth'].shape, (3,))
+        self.assertApproxEqual(ds['refBottomDepth'][-1],
+                               4.882000207901)
+        date = pd.Timestamp(ds.Time.values[0])
+        # round to nearest second
+        date = pd.Timestamp(long(round(date.value, -9)))
+        self.assertEqual(date, pd.Timestamp('1855-01-13 12:24:14'))
+
+
+    def test_no_units(self):
+        fileName = str(self.datadir.join('example_no_units_jan.nc'))
+        timestr = 'time_avg_daysSinceStartOfSim'
+        varList = \
+            ['time_avg_avgValueWithinOceanLayerRegion_avgLayerTemperature',
+             'refBottomDepth']
+
+        ds = xr.open_mfdataset(
+            fileName,
+            preprocess=lambda x: mpas_xarray.preprocess_mpas(x,
+                                                             timestr=timestr,
+                                                             onlyvars=varList,
+                                                             yearoffset=1850))
+        self.assertEqual(ds.data_vars.keys(), varList)
+        date = pd.Timestamp(ds.Time.values[0])
+        # round to nearest second
+        date = pd.Timestamp(long(round(date.value, -9)))
+        self.assertEqual(date, pd.Timestamp('1855-01-13 12:24:14'))
+
+
+#    def test_selvals(self):
```

Comment:
@pwolfram, I tried to make a unit test for using `selvals` but (as you can see by uncommenting this function) it didn't work as expected.  For some reason, xarray doesn't seem to recognize that `refBottomDepth` varies along dimension `nVertLevels` so it tries to treat `refBottomDepth` itself as a coordinate (which it is not) and fails.  If you find a fix, please push it to this branch!